lib/backend/k8s/k8s.go

```diff
@@ -142,14 +142,20 @@ func NewKubeClient(kc *capi.KubeConfig) (*KubeClient, error) {
 	return kubeClient, nil
 }
 
+// EnsureInitialized checks that the necessary custom object definitions
+// exist in the backend. This usually passes when using etcd
+// as a backend but can often fail when using KDD as it relies
+// on various custom resources existing.
+// To ensure the datastore is initialized, this function checks that a
+// known custom resource is defined: FelixGlobalConfigSetting. It accomplishes this
+// by trying to set the ClusterType (an example of FelixGlobalConfigSetting).
```

Comment:
not sure I understand what you mean by `an example of FelixGlobalConfigSetting`. `ClusterType` is an instance of `GlobalFelixConfig` CRD